### DISCUSSION TITLE
How can I hide messages with the `LSP[metals]` prefix?

### DISCUSSION THREAD

**Discussion by FelipeLema** [scalameta/nvim-metals#613] [Q&A] [answered]
Hey, there

How can I hide messages prefixed with `LSP[metals]` that are printed in the commandline? Example:

```
LSP[metals][Log] Importing build⠇
LSP[metals][Log] Indexing⠇
LSP[metals][Log] Indexing⠋
LSP[metals][Log] Indexing⠙
LSP[metals][Log] Indexing⠸
LSP[metals][Log]  Indexing complete!
LSP[metals][Log] Downloading scalafix⠇
LSP[metals][Log] Compiling myProject   
```

**Comment by ckipp01** [comment]
Hey @FelipeLema I'm pretty sure this party is simply coming right from Neovim. This is how it handles `window/logMessage` notifications, and displays this this way. You might be able to change this by overwriting the default handler. With that being said, if you're seeing these, then you're not enabling metals status. I highly recommend doing that as the logs you'll receive will be much more precise and they won't fill up your commandline like this. All you need to do is capture it in your status line.

**Comment by FelipeLema** [comment]
hey, thanks @ckipp01, but setting either or both still is idempotent and I'm still getting those messages printed... could I be missing something else?

- override `"window/logMessage" handler
```lua
local ms = require('vim.lsp.protocol').Methods or {window_logMessage = "window/logMessage"}
local window_logMessage_handler = vim.lsp.handlers[ms.window_logMessage]
vim.lsp.handlers[ms.window_logMessage] = function(err, result, ctx, config)
  local cliente = vim.lsp.get_client_by_id(ctx.client_id)
  if cliente.name == "metals" then
    return result
  end
  return window_logMessage_handler(err, result, ctx, config)
end
```

- turn on status
```lua
require 'metals.handlers'["metals/status"] = function(...) end
```

**Reply by ckipp01** [reply, marked as answer]
> turn on status

All you actually need to do to turn this on is outlined [here](https://github.com/scalameta/nvim-metals/blob/f41c14ae8500ceccb71e6695574b67881f0b5a93/doc/metals.txt#L101-L110).
